src/components/Modal/index.web.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import withWindowDimensions from '../withWindowDimensions';
+import BaseModal from './BaseModal';
+import {propTypes, defaultProps} from './modalPropTypes';
+import * as StyleUtils from '../../styles/StyleUtils';
+
+const Modal = props => (
+    <BaseModal
+            // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+        onModalHide={() => {
+            if (props.fullscreen) {
+                // With the modal hidden, we can revert to the default status bar color (refer to https://github.com/Expensify/App/issues/12156).
+                document.querySelector('meta[name=theme-color]').content = '';
+            }
+
+            props.onModalHide();
+        }}
```

Comment:
To follow our code conventions better, could you please move this logic into function definitions at the top of the render function? Kinda like this (just a rough idea):

```
(props) => {
    const setStatsuBarColor = function (color = '') {
        blah blah
    };
    return (<BaseModal onModalHide={setStatsuBarColor} />);
};
```

Reply:
You can also just use a single method and have the color passed as an argument

Reply:
Done! Though, we still need two functions to call `props.onModalHide()` and `props.onModalShow()`, respectively.

Reply:
Ah yes, you're right. Still need both of them. Thanks!